receiver/awsfirehosereceiver/receiver_test.go

```diff
@@ -159,13 +158,6 @@ func TestFirehoseRequest(t *testing.T) {
 			wantStatusCode: http.StatusInternalServerError,
 			wantErr:        firehoseConsumerErr,
 		},
-		"WithCorruptBase64Records": {
```

Comment:
Why remove this?

Reply:
This test meant: I send one corrupt record and there is no error. And this is no longer valid, because now the metrics/logs that have count 0, have an error. It is still possible to send records with both valid and invalid format (all unmarshallers have those tests), but it is not possible to send records and end up with 0 metrics and logs.

Reply:
Sorry, my first reply to this was wrong. Before this PR, if there was a wrong record, then everything else would fail. Now this does not happen anymore (see [this line](https://github.com/constanca-m/opentelemetry-collector-contrib/blob/1d0432c9b9348ca3d2b2b8090b75eafcb892803d/receiver/awsfirehosereceiver/receiver.go#L264-L272)), since we have a `continue` statement instead of throwing an error.

I did this because we follow the same logic for all unmarshallers, and I wanted to keep it consistent here too.

Reply:
Ah I see. I was thinking of proposing that unmarshallers stop ignoring errors - I'm not aware that other receivers that behave like this, but maybe just out of ignorance. At any rate, I'd prefer that we don't expand this behaviour. If it's not strictly necessary for this PR, can we revert to the original behaviour, and keep it for another PR?

Reply:
Yes, I have reverted it now.